internal/pkg/template/templates/cicd/pipeline_cfn.yml

```diff
@@ -124,6 +124,14 @@ Resources:
           {{- end }} {{/* endif ne .Source.OutputArtifactFormat "" */}}{{- end }} {{/* endif ne .Source.ProviderName "GitHubV1" */}}
       Roles:
         - !Ref BuildProjectRole
+  {{- if .Build.AdditionalPolicy }}
+  BuildProjectAdditionalPolicy:
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyName: !Sub ${AWS::StackName}-BuildProjectAdditionalPolicy
+  {{indent 4 .Build.AdditionalPolicy}}
```

Comment:
If I'm understanding correctly, this is assume that clients will write the pipeline manifest like this:
```yaml
build:
  additional_policy:
    PolicyDocument:
      Statement: ...
```

It's not super clear why `Roles` and `PolicyName` cannot be set in `additional_policy`. Can we instead change the struct to be something like this?
```go
type Build struct {
   // ..
   AdditionalPolicy struct {
      Document yaml.Node `yaml:"document",omitempty`
   } `yaml:"additional_policy,omitempty"`
}
```
The above makes it clear to me that only `PolicyDocument` will be customizable by clients

Reply:
Yeah this looks clean, I will change it.